charts/redpanda/templates/statefulset.yaml

```diff
@@ -37,23 +39,16 @@ metadata:
 {{- end }}
 spec:
   selector:
-    matchLabels:
-      app.kubernetes.io/name: {{ template "redpanda.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
+    matchLabels: {{ (include "statefulset-pod-labels" .) | nindent 6 }}
   serviceName: {{ template "redpanda.fullname" . }}
   replicas: {{ .Values.statefulset.replicas | int64 }}
   updateStrategy:
     {{- toYaml .Values.statefulset.updateStrategy | nindent 4 }}
   podManagementPolicy: {{ .Values.statefulset.podManagementPolicy }}
   template:
     metadata:
-      labels:
-        app.kubernetes.io/name: {{ template "redpanda.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name | quote }}
-        app.kubernetes.io/component: {{ template "redpanda.name" . }}
-{{- with .Values.commonLabels }}
-        {{- toYaml . | nindent 8 }}
-{{- end }}
+      labels: {{ (include "statefulset-pod-labels" .) | nindent 8 }}
+        redpanda.com/poddisruptionbudget: {{ template "redpanda.name" . }}
```

Comment:
Why `redpanda.com/poddisruptionbudget` is not included in `_statefulset.tpl` file under `"statefulset-pod-labels"` template?

Reply:
Because we _can_ change labels, but we cannot change the selector. Since we want to use the same template for labels and selector, but existing installs are broken, this at least allows the PDB to only target the STS pods since the PDB selector _is_ mutable.